sopel/bot.py

```diff
@@ -648,3 +646,71 @@ def cap_req(self, module_name, capability, arg=None, failure_callback=None,
             entry.append(_CapReq(prefix, module_name, failure_callback, arg,
                                  success_callback))
             self._cap_reqs[cap] = entry
+
+    def register_url_callback(self, pattern, callback):
+        """Register a ``callback`` for URLs matching the regex ``pattern``
+
+        :param pattern: compiled regex pattern to register
+        :param callback: callable object to handle matching URLs
+
+        .. versionadded: 7.0.0
+
+            This method has been added in Sopel 7.0.0 in order to offer
+            a unified interface for URL callbacks.
```

Comment:
Redundant to say "This method has been added in Sopel 7.0.0"—the `versionadded` directive already says that for us. (Applies to all three added docstrings.)

This explanatory text is OK, but I think it would be more useful to show module authors what code it replaces in their `setup` functions (the old `bot.memory` check and subsequent manual assignment into the dict).

Reply:
Done.